src/makeInfinityAdder.js

```diff
@@ -4,7 +4,25 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  const adder = (number) => {
+    if (number === undefined) {
+      const result = sum;
+
+      sum = 0;
+
+      return result;
+    }
+
+    sum += number;
+
+    if (number !== undefined) {
+      return adder;
+    }
```

Comment:
The check for `number !== undefined` is redundant here because if the code execution has reached this point, it means that `number` was not `undefined`. You can safely remove this if statement.